utils/ccdatofhir/mapping_datatypes.bal

```diff
@@ -0,0 +1,205 @@
+#
+# + idElement - C-CDA id element
+# + return - Return FHIR Identifier
+public isolated function mapCcdaIdToFhirIdentifier(xml idElement) returns r4:Identifier? {
+    r4:Identifier identifier;
+    string?|error idVal = idElement.extension;
+    string?|error rootVal = idElement.root;
+    if rootVal is string {
+        identifier = {};
+        if rootVal.equalsIgnoreCaseAscii("2.16.840.1.113883.4.1") && idVal is string {
+            identifier.system = "http://hl7.org/fhir/sid/us-ssn";
+            identifier.value = idVal;
+        } else if idVal is string {
+            identifier.system = rootVal;
+            identifier.value = string `urn:oid:${idVal}`;
+        } else {
+            identifier.system = "urn:ietf:rfc:3986";
+            identifier.value = rootVal;
+        }
+        return identifier;
+    } else {
+        log:printDebug("Mandatory field root not available", rootVal);
+        return ();
+    }
+}
+
+# Map C-CDA address to FHIR Address.
+#
+# + addressElement - C-CDA address element
+# + return - Return FHIR Address
+public isolated function mapCcdaAddressToFhirAddress(xml addressElement) returns r4:Address? {
+    r4:Address address = {};
+    xml streetAddressLineElement = addressElement/<v3:streetAddressLine|streetAddressLine>;
+    xml cityElement = addressElement/<v3:city|city>;
+    xml stateElement = addressElement/<v3:state|state>;
+    xml postalCodeElement = addressElement/<v3:postalCode|postalCode>;
+
+    string streetAddressLine = streetAddressLineElement.data().trim();
+    string city = cityElement.data().trim();
+    string state = stateElement.data().trim();
+    string postalCode = postalCodeElement.data().trim();
+
+    if streetAddressLine != "" || city != "" || state != "" || postalCode != "" {
+        address.line = [streetAddressLine];
+        address.city = city;
+        address.state = state;
+        address.postalCode = postalCode;
+        return address;
+    } else {
+        log:printDebug("Address fields not available");
+        return ();
+    }
+}
+
+# Map C-CDA telecom to FHIR ContactPoint.
+#
+# + telecomElement - C-CDA telecom element
+# + return - Return FHIR ContactPoint
+public isolated function mapCcdaTelecomToFhirTelecom(xml telecomElement) returns r4:ContactPoint? {
+    r4:ContactPoint telecom = {};
+    string?|error telecomUse = telecomElement.use;
+    string?|error telecomValue = telecomElement.value;
+
+    if telecomValue is string {
+        telecom.value = telecomValue;
+    } else {
+        log:printDebug("Telecom value not available", telecomValue);
+    }
+
+    if telecomUse is string {
+        match telecomUse {
+            "HP" => {
+                telecom.use = "home";
+            }
+            "WP" => {
+                telecom.use = "work";
+            }
+        }
+    } else {
+        log:printDebug("Telecom use not available", telecomUse);
+    }
+
+    if telecom.use is string && telecom.value is string{
```

Comment:
```suggestion
    if telecom.use is string && telecom.value is string {
```

Check formatting for all the changes.

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/cc257cbf67919d7d86a4e90e88ec03e1250eadcf